website/docs/API.md

```diff
@@ -515,22 +517,94 @@ Defined as:
+
+```tsx
+// ❌ won't throw because callback never throws
+await waitFor(() => false);
+```
+
+`waitFor` only rejects when it reaches its timeout so if you don't await it it will never throw.
 
 ```jsx
-import { render, screen, waitFor } from '@testing-library/react-native';
+// ❌ won't throw because waitFor is not awaited
+waitFor(() => expect(1).toBe(2))
+```
 
-test('waiting for an Banana to be ready', async () => {
-  render(<Banana />);
+:::note
+You can enforce awaiting waitFor by using the [await-async-utils](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/await-async-utils.md) rule from [eslint-plugin-testing-library](https://github.com/testing-library/eslint-plugin-testing-library).
+:::
 
-  await waitFor(() => screen.getByText('Banana ready'));
-});
+Since `waitFor` is likely to run `expectation()` multiple times, it is [highly recommended not to have any side effects](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#performing-side-effects-in-waitfor) in `waitFor`.
+
+```jsx
+await waitFor(() => {
+  // ❌ button will be pressed on each waitFor iteration
+  fireEvent.press(screen.getByText('press me'))
+  expect(mockOnPress).toHaveBeenCalled()
+})
+```
+
+:::note
+Not using fireEvent in `waitFor` can be enforced with the rule [no-wait-for-side-effects](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-wait-for-side-effects.md).
+:::
+
+It is also recommended to have a [single expectation per waitFor](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#having-multiple-assertions-in-a-single-waitfor-callback) for more consistency and faster failing tests. If you want to make several assertions, then they should be in seperated `waitFor`, or maybe the second assertion won't even require a `waitFor`.
+
+### With jest fake timers
+
+`waitFor` checks wether jest fake timers are enabled and adapts its behavior when it is the case. The following snippet is a simplified version of how it behaves when fake timers are enabled:
+
+```tsx
+let fakeTimeRemaining = timeout;
+
+while(fakeTimeRemaining > 0) {
+  fakeTimeRemaining = fakeTimeRemaining - interval;
+  jest.advanceTimersByTime(interval);
+  try {
+    expectation();
+    // resolve
+  }
+}
+
+// reject
+```
+
+This implementation doesn't depend on real time and automatically advances fake timers.
+
+```tsx
+// in component
+setTimeout(() => {
+  someFunction();
+}, 10000)
+
+
+// in test
+jest.useFakeTimers();
+
+await waitFor(() => {
+  expect(someFunction).toHaveBeenCalledWith();
+}, 10000)
 ```
 
+:::note
+If you want to wait for a long period of time, it can result in a very long test, especially when it is costly to run `expectation()`. In those cases, you can increase the `interval` as well so that you don't perform as many checks.
```

Comment:
This comments seems confusing, not sure it relates to waiting under real or fake timers? It also seems quite technical and low level. Pls clarify it or consider removing it.

Reply:
I don't see many cases where expectation was long to run, have you encountered such a case before?
if not, I vote for removing it as well

Reply:
It relates to having fake timers, since when using real timers it will depend on real time. I have encountered this situation before yes, if you're testing let's say a timeout of 10 minutes or maybe more and keep the default 50ms interval you're going to run quite a lot of queries without it being useful. It's more of an edge case though so I removed  it, maybe it should be mentioned in some advanced section but I'll leave it out for now